### PR TITLE
Add Reducers and Action

### PR DESCRIPTION
**Pull request by hetrox8**
This branch's main feature is to configure the Project with redux react toolkit.
The main languages used include:
1. Redux toolkit
2. React
3. Javascript

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      state.books.push(action.payload);
+    },
+    removeBook: (state, action) => {
+      state.books = state.books.filter((book) => book.id !== action.payload);
+    },
+  },
+});
+
+export const { addBook, removeBook } = BooksSlice.actions;
+export default BooksSlice.reducer;
```
Comment (AmaduKamara) [inline src/redux/categories/categoriesSlice.js]: - [ ] I highly recommend you change the name of this file to `categoriesSlice` and not BooksSlice.  
- [ ] For this file, you're expected to have one method `checkStatus` that returns the text `Under construction` and not these reducers. Kindly remove these reducers and create 1 reducer - checkStatus in this file as required 👍🏽
- [ ] Let's export the checkStatus reducer after you implement it and the actions as well 👍🏽

File: src/redux/book/bookSlice.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+const BooksSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    AddBook: (state, action) => {
+      state.books.push(action.payload);
+    },
+    RemoveBook: (state, action) => {
+      state.books = state.books.filter((book) => book.id !== action.payload);
+    },
```
Comment (AmaduKamara) [inline src/redux/book/bookSlice.js]: ## Optional Change ♻️
- [ ] It's a good convention to have our reducer function names start with a lowercase letter. For instance, the AddBook should be written as `addBook`. Kindly modify your reducer function names 👍🏽

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @hetrox8,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/redux/book/bookSlice.js (hunk 2 above)
Comment (hetrox8) [reply src/redux/book/bookSlice.js:16]: you got to be kidding me bruh.

Comment (AmaduKamara) [reply src/redux/book/bookSlice.js]: It's not a required change though, but it's a good convention to start a function name in lowercase letters. 👍🏽

Comment (AmaduKamara) [reply src/redux/book/bookSlice.js]: 👉🏼 removeBook: (state, action) => {
      state.books = state.books.filter((book) => book.id !== action.payload);
    },

**Comment by hetrox8** [comment]
1.change name of file 
2.change file to return text under construction
3.change reducer function to  lowercase

File: src/redux/categories/categoriesSlice.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      category: 'Personal Development',
+    },
+    {
+      id: 2,
+      title: 'Personal Development',
+      author: 'Jim Rohn',
+      category: 'Personal Development',
+    },
+  ],
+};
```
Comment (AbelG101) [inline src/redux/categories/categoriesSlice.js:18]: - [ ] [OPTIONAL] The previous reviewer forgot to mention this but I'd kindly suggest you carefully read the project requirements before making changes to your code base as it's specifically stated that the initial state of the categories should be an empty array as shown below 👇, not an array of books.
```jsx
initialState: {
    categories: [],
  },
```

File: src/redux/categories/categoriesSlice.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    checkStatus: () => 'Under construction',
+  },
+});
```
Comment (AbelG101) [inline src/redux/categories/categoriesSlice.js:26]: - [ ] The previous reviewer forgot to mention this as well but as part of the project requirements please make sure to check the action payload before returning the message as shown below, this means you'll have to add parameters `state` and `action` as this is a reducer
```jsx
checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },

```

File: src/redux/categories/categoriesSlice.js
Comment (AbelG101) [inline src/redux/categories/categoriesSlice.js:1]: - [ ] **Please carefully revise the suggestions as your app will not work properly and cause for rework in the next milestones.**

**Review by AbelG101 — APPROVED** [review]
# Hi @hetrox8 👋, 

**Your project is complete!  There is nothing else to say other than... it's time to merge it. 
Congratulations! 🎉 🥳🥇**

## Highlights
- Correct use of Git Flow ✔
-  Well-structured files ✔
- All linter checks are passing ✔
- Descriptive PR title with a good summary ✔

## Optional suggestions
- CHECK THE COMMENTS BELOW.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**